ext/imap/php_imap.stub.php

```diff
@@ -0,0 +1,354 @@
+<?php
+
+/**
+ * @return resource|false
+ */
+function imap_open(string $mailbox, string $user, string $password, int $options = 0, int $n_retries = 0) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_reopen($stream_id, string $mailbox, int $options = 0, int $n_retries = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_close($stream_id, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return int|false
+ */
+function imap_num_msg($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return int|false
+ */
+function imap_num_recent($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_headers($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_headerinfo($stream_id, int $msg_no, int $from_length = 0, int $subject_length = 0, string $default_host = '') {}
+
+function imap_rfc822_parse_headers(string $headers, string $default_host = ''): object {}
+
+/**
+ * @return string|false
+ */
+function imap_rfc822_write_address(string $mailbox, string $host, string $personal) {}
+
+function imap_rfc822_parse_adrlist(string $address_string, string $default_host): array {}
+
+
+/**
+ * @param resource $stream_id
+ * @return string|false
+ */
+function imap_body($stream_id, int $msg_no, int $options = 0) {}
+
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_bodystruct($stream_id, int $msg_no, string $section) {}
+
+/**
+ * @param resource $stream_id
+ * @return string|false
+ */
+function imap_fetchbody($stream_id, int $msg_no, string $section, int $options = 0) {}
+
+/**
+ * @param resource $stream_id
+ * @return string|false
+ */
+function imap_fetchmime($stream_id, int $msg_no, string $section, int $options = 0) {}
+
+
+/**
+ * @param resource $stream_id
+ * @todo: should $file be `resouce|string`? it looks like it tries to accept anything?
+ */
+function imap_savebody($stream_id, $file, int $msg_no, string $section = '', int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return string|false
+ */
+function imap_fetchheader($stream_id, int $msg_no, int $options = 0) {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_fetchstructure($stream_id, int $msg_no, int $options = 0) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_gc($stream_id, int $flags): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_expunge($stream_id): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_delete($stream_id, int $msg_no, int $options = 0): bool {}
```

Comment:
`$msg_no` is a zval which is converted to string, so I think:
```suggestion
function imap_delete($stream_id, string $msg_no, int $options = 0): bool {}
```